.github/workflows/design-book.yml

```diff
@@ -0,0 +1,35 @@
+name: Design book
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: "latest"
+
+      - name: Build
+        working-directory: ./design
+        run: mdbook build
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        # Only deploy if pushed to the main branch
+        if: ${{ github.ref == 'refs/heads/main' }}
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
What is the purpose of using the GITHUB_TOKEN here?

Reply:
It seems to be necessary for the workflow to authenticate itself, I assume to push to the `gh-pages`branch.

From [the docs](https://github.com/peaceiris/actions-gh-pages):

> For newbies of GitHub Actions: Note that the `GITHUB_TOKEN` is **NOT** a personal access token. A GitHub Actions runner automatically creates a `GITHUB_TOKEN` secret to authenticate in your workflow. So, you can start to deploy immediately without any configuration.